relay/README.md

```diff
@@ -149,26 +148,42 @@ and the eventual closing of the stream.
  +------------ multiaddr of the dialing node
 ```
 
-After getting a stream to the relay node from its libp2p swarm,
-the dialing transport writes the header to the stream.
-The relaying node reads the header, gets a stream to the destination node,
-then writes the header to the destination stream and shortcircuits the two streams.
+After getting a stream to the relay node from its libp2p swarm, the dialing transport writes the header to the stream.
+
+The relaying node reads the header, gets a stream to the destination node, then writes the header to the destination stream and shortcircuits the two streams.
 
-Each relayed connection corresponds to two multistreams,
-one between QmOne and QmRelay, the other between QmRelay and QmTwo.
+Each relayed connection corresponds to two multistreams, one between QmOne and QmRelay, the other between QmRelay and QmTwo.
 
 Implementation details:
 - The relay node has the `Swarm.EnableRelaying` config option enabled
 - The relay node allows only one relayed connection between any two nodes.
 - The relay node validates the `src` header field.
 - The listening node validates the `dst` header field.
 
+### Error table
+
+This is a table of error codes and sample messages that may occur during a relay setup. Codes in the 200 range are returned by the relay node. Codes in the 300 range are returned by the destination node.
+
+
+| Code  | Message           | Meaning    |
+| ----- |:-----------------:|:----------:|
+| 100   | OK                      | Relay was setup correctly    |
+| 220   | "src address too long"  | |
+| 221   | "dst address too long"  | |
+| 250   | "failed to parse src addr: no such protocol ipfs" | The `<src>` multiaddr in the header was invalid |
+| 251   | "failed to parse dst addr: no such protocol ipfs" | The `<dst>` multiaddr in the header was invalid |
+| 260   | "passive relay has no connection to dst" | |
+| 261   | "active relay could not connect to dst: connection refused" | relay could not form new connection to target peer |
+| 262   | "could not open new stream to dst: BAD ERROR" | relay has connection to dst, but failed to open a new stream |
+| 270   | "<dst> does not support relay" | |
+| 320   | "src address too long" | |
```

Comment:
From IRC:

```
14:13 <@daviddias> I don't understand why the 'send the src' addr keeps appearing as a thing for relay
16:05 <+dryajov> diasdavid re srs addr: I dont think we need it anymore, you are right. We can always do getPeerInfo on the circuited conn and should be able to get the src addrs. It would only make sense, if the full relay addr is sent so that the relay can be rebuilt from the dst to the src, if that ever makes sense?
16:23 <@daviddias> " It would only make sense, if the full relay addr is sent so that the relay can be rebuilt from the dst to the src, if that ever makes sense?" even in that case, we can always verify who dialed us with the getPeerInfo call
16:26 <+dryajov> True, that would be a swarm addr and should come back with the getPeerInfo. We don't need it.
```